expertise/service/expertise.py

```diff
@@ -0,0 +1,303 @@
+        config['job_dir'] = root_dir
+        config['cdate'] = int(time.time())
+        config['status'] = JobStatus.INITIALIZED.value
+        config['description'] = descriptions[JobStatus.INITIALIZED]
+
+        # Set SPECTER+MFR paths
+        if config.get('model', 'specter+mfr') == 'specter+mfr':
+            config['model_params']['specter_dir'] = self.specter_dir
+            config['model_params']['mfr_feature_vocab_file'] = self.mfr_feature_vocab_file
+            config['model_params']['mfr_checkpoint_dir'] = self.mfr_checkpoint_dir
+
+        # Create directory and config file
+        if not os.path.isdir(config['dataset']['directory']):
+            os.makedirs(config['dataset']['directory'])
+        with open(os.path.join(root_dir, 'config.json'), 'w+') as f:
+            ## Remove the token before saving this in the file system
+            token = config.get('token', None)
+            if token is not None:
+                del config['token']
+                json.dump(config, f, ensure_ascii=False, indent=4)
+                config['token'] = token
+            else:
+                json.dump(config, f, ensure_ascii=False, indent=4)
+
+        return config
+
+    def _get_subdirs(self, user_id=None):
+        """
+        Returns the direct children directories of the given root directory
+
+        :returns: A list of subdirectories not prefixed by the given root directory
+        """
+        subdirs = [name for name in os.listdir(self.working_dir) if os.path.isdir(os.path.join(self.working_dir, name))]
+        if user_id is None:
+            return subdirs
+        else:
+            # If given a profile ID, assume looking for job dirs that contain a config with the
+            # matching profile id
+            filtered_dirs = []
+            for subdir in subdirs:
+                config_dir = os.path.join(self.working_dir, subdir, 'config.json')
+                with open(config_dir, 'r') as f:
+                    config = json.load(f)
+                    if user_id == config['user_id'] or user_id.lower() in SUPERUSER_IDS:
+                        filtered_dirs.append(subdir)
+            return filtered_dirs
+
+    def _get_score_and_metadata_dir(self, search_dir):
+        """
+        Searches the given directory for a possible score file and the metadata file
+
+        :param search_dir: The root directory to search in
+        :type search_dir: str
+
+        :returns file_dir: The directory of the score file, if it exists, starting from the given directory
+        :returns metadata_dir: The directory of the metadata file, if it exists, starting from the given directory
+        """
+        # Search for scores files (only non-sparse scores)
+        file_dir, metadata_dir = None, None
+        with open(os.path.join(search_dir, 'config.json'), 'r') as f:
+            config = json.load(f)
+        # Look for score files
+        for root, dirs, files in os.walk(search_dir, topdown=False):
```

Comment:
Why are you searching the whole directory? You can directly look for file using `os.path.exists`